docs/Spark/gems/source-target/file/iceberg.md

```diff
@@ -0,0 +1,193 @@
+
+   ![Open Pipeline Settings](./img/iceberg/open-pipeline-settings.png)
+
+2. On the Spark dialog, under Spark Configuration, add the Spark conf properties.
+
+   ![Spark Pipeline Settings](./img/iceberg/spark-pipeline-settings.png)
+
+## Source
+
+### Source Parameters
+
+| Parameter                   | Description                                                                    | Required                                                                     |
+| --------------------------- | ------------------------------------------------------------------------------ | ---------------------------------------------------------------------------- |
+| Catalog Name                | Any configured Hadoop/Hive catalog name                                        | True (If any default catalog is not configured in Spark runtime properties.) |
+| Schema Name (Database Name) | Name of the database                                                           | True                                                                         |
+| Table Name                  | Name of the table                                                              | True                                                                         |
+| Read Timestamp              | Time travel to a specific timestamp (value should be in milliseconds)          | False                                                                        |
+| Read Snapshot               | Time travel to a specific version of the table (value should be a snapshot ID) | False                                                                        |
+
+:::note
+
+For time travel on Iceberg tables:
+
+1. Only `Read Timestamp` **_OR_** `Read Snapshot` can be selected, not both.
+2. Timestamp should be between the first commit timestamp and the latest commit timestamp in the table.
+3. Snapshot needs to be a snapshot ID.
+
+By default most recent version of each row is fetched if no time travel option is used.
+
+:::
+
+:::info
+
+To read more about Iceberg time travel and its use cases, see the [Apache Iceberg docs](https://iceberg.apache.org/docs/nightly/hive/?h=time#timetravel).
+
+:::
+
+### Example {#source-example}
+
+<div class="wistia_responsive_padding" style={{padding:'56.25% 0 0 0', position:'relative'}}>
+<div class="wistia_responsive_wrapper" style={{height:'100%',left:0,position:'absolute',top:0,width:'100%'}}>
+<iframe src="https://fast.wistia.net/embed/iframe/b1vt6gv6bl?seo=false?videoFoam=true" title="Iceberg source example" allow="autoplay; fullscreen" allowtransparency="true" frameborder="0" scrolling="no" class="wistia_embed" name="wistia_embed" msallowfullscreen width="100%" height="100%"></iframe>
+</div></div>
+<script src="https://fast.wistia.net/assets/external/E-v1.js" async></script>
+
+### Generated Code {#source-code}
+
+````mdx-code-block
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs>
+
+<TabItem value="py" label="Python">
+
+```py
```

Comment:
Is it possible to get the code example in Scala too?

Reply:
Scala implementation is yet not there.